From 469571728098c94c937865548df0b049a5395503 Mon Sep 17 00:00:00 2001
From: Pavel Emelyanov <xemul@parallels.com>
Date: Thu, 17 Feb 2011 12:12:23 +0300
Subject: [PATCH 5/6] ve: Fix d_path return code when no buffer given

In case we call d_path with NULL buffer (d_root_check) the
return is *always* NULL, which is not what we actually want...

Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 fs/dcache.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/fs/dcache.c b/fs/dcache.c
index 7fce87d..dc638a3 100644
--- a/fs/dcache.c
+++ b/fs/dcache.c
@@ -2037,7 +2037,7 @@ out:
 	if (deleted && buffer &&
 			prepend(&end, &buflen, " (deleted)", 10) != 0)
 		goto Elong;
-out_err:
+
 	spin_unlock(&vfsmount_lock);
 	return buffer ? retval : NULL;
 
@@ -2070,7 +2070,10 @@ global_root:
 
 Elong:
 	retval = ERR_PTR(-ENAMETOOLONG);
-	goto out_err;
+out_err:
+	spin_unlock(&vfsmount_lock);
+	return retval;
+
 }
 EXPORT_SYMBOL(__d_path);
 
-- 
1.7.4.1

